Disable guided onboarding on serverless

alisonelizabeth opened the pull request:
This PR disables guided onboarding on serverless (not being used), and thus makes the `guided_onboarding` plugin an optional dependency.

Note: Discovered https://github.com/elastic/kibana/issues/168401 while testing

## How to test
Please test on both stateful and serverless.

### On stateful
All onboarding guides should work as before. **If you are an owner of one of the following E2E guides, please test to confirm everything is still working as expected**.
- Kubernetes ("Monitor Kubernetes clusters")
- SIEM ("Detect threats in my data with SIEM")
- Search ("Build search experiences with Elasticsearch", "Search across databases and business systems", "Add search to my website" - all 3 guides follow the same flow, but please confirm each one)

### On serverless
Serverless projects should work as before, however, there were a lot of places in the code where we were assuming `guidedOnboarding` was defined. **My ask is that each solution tests their project to make sure there are no issues.** 

**Other things to verify:**
 
- The "help" menu in the main header should no longer show the "Setup guides" link
- User should not be able to access the `app/home#/getting_started` route
- There should be no requests to `GET /internal/guided_onboarding/state`

alisonelizabeth committed:
disable guided onboarding on serverless

alisonelizabeth committed:
disable guided onboarding on serverless

alisonelizabeth committed:
disable guided onboarding on serverless

alisonelizabeth committed:
disable guided onboarding on serverless

alisonelizabeth committed:
disable guided onboarding on serverless

alisonelizabeth committed:
disable guided onboarding on serverless

alisonelizabeth committed:
Merge branch 'main' of github.com:elastic/kibana into disable_guided_onboarding

alisonelizabeth committed:
revert ordering changes

alisonelizabeth committed:
Merge branch 'main' of github.com:elastic/kibana into disable_guided_onboarding

alisonelizabeth commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into disable_guided_onboarding

alisonelizabeth committed:
fix security project

alisonelizabeth committed:
Merge branch 'main' of github.com:elastic/kibana into disable_guided_onboarding

alisonelizabeth committed:
make sure guidedOnboarding plugin is optional

alisonelizabeth commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into disable_guided_onboarding

alisonelizabeth committed:
Merge branch 'main' of github.com:elastic/kibana into disable_guided_onboarding

alisonelizabeth committed:
make guidedOnboarding optional for Observability plugin

alisonelizabeth commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into disable_guided_onboarding

alisonelizabeth commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into disable_guided_onboarding

alisonelizabeth committed:
Merge branch 'main' of github.com:elastic/kibana into disable_guided_onboarding

yuliacech commented on x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/search_index.tsx:
```diff
@@ -80,24 +78,39 @@ export const SearchIndex: React.FC = () => {
+    const subscription = guidedOnboarding?.guidedOnboardingApi
+      ?.isGuideStepActive$('appSearch', 'add_data')
+      .subscribe((isStepActive) => {
+        if (isStepActive) {
```
I think the check before also included `index.count`
```suggestion
        if (isStepActive && index?.count) {
```

alisonelizabeth replied:
Nice catch! Fixed.

yuliacech commented on x-pack/plugins/exploratory_view/public/plugin.ts:
```diff
@@ -60,7 +60,7 @@ export interface ExploratoryViewPublicPluginsStart {
   embeddable: EmbeddableStart;
 
-  guidedOnboarding: GuidedOnboardingPluginStart;
+  guidedOnboarding?: GuidedOnboardingPluginStart;
```
I know this dependency was there before, but it seems to be unused in this plugin. Probably a copy-paste typo from https://github.com/elastic/kibana/pull/153852

alisonelizabeth replied:
Another great catch! You're right, it does not appear to be used, so I removed it.

yuliacech commented on examples/guided_onboarding_example/public/components/step_one.tsx:
```diff
@@ -23,27 +23,25 @@ import {
   useEffect(() => {
-    setIsTourStepOpen(isTourActive);
-  }, [isTourActive]);
+    const subscription = guidedOnboarding?.guidedOnboardingApi
```
I noticed that we still have a couple of places in the examples plugin that still mention `observable`. One is in the component `main`, but this was even before this PR. And the other mention is in this component, I think we could re-phrase those texts.

alisonelizabeth replied:
I updated the copy to say an observable subscription; but happy to change to something else if you think it's still confusing

alisonelizabeth committed:
Merge branch 'main' of github.com:elastic/kibana into disable_guided_onboarding

alisonelizabeth committed:
address initial review feedback

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/lint_ts_projects --fix'

alisonelizabeth commented on x-pack/plugins/exploratory_view/kibana.jsonc:
```diff
@@ -15,7 +15,6 @@
       "dataViews",
       "features",
       "files",
-      "guidedOnboarding",
```
Does not appear to be used anywhere so removed it as a dependency

elasticmachine commented:
Pinging @elastic/platform-onboarding (Team:Journey/Onboarding)

pgayvallet reviewed (approved):
LGTM for serverless config file

elasticmachine commented:
Pinging @elastic/fleet (Team:Fleet)

criamico reviewed (approved):
Fleet changes LGTM

Dosant reviewed (approved):
src/plugins/home/public/application/components/home_app.js lgtm

sphilipse reviewed (approved):
search changes LGTM

yuliacech commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into disable_guided_onboarding

yuliacech committed:
Merge branch 'main' into disable_guided_onboarding

cauemarcondes reviewed (approved):
Obs changes LGTM

bollinic reviewed (approved):
onboarding changes looks good

yuliacech committed:
Merge branch 'main' into disable_guided_onboarding

stephmilovic reviewed (commented):
Could you also please update our cypress tests? [This line](https://github.com/elastic/kibana/blob/main/x-pack/test/security_solution_cypress/cypress/e2e/explore/guided_onboarding/tour.cy.ts#L33) needs to be updated from `tags: ['@ess', '@brokenInServerless']` to `tags: ['@ess']`

alisonelizabeth committed:
Merge branch 'main' of github.com:elastic/kibana into disable_guided_onboarding

alisonelizabeth committed:
update tags for cypress tests

yuliacech reviewed (commented):
I tried to run the SIEM guide through but got stuck on the Rules step. @xcrzx, could you please check that step in the SIEM guide works correctly? I looked at an 8.11 build candidate deployment and there the UI looks the same, it's unclear how to get the step completed. 
<img width="1495" alt="Screenshot 2023-10-23 at 19 33 31" src="https://github.com/elastic/kibana/assets/6585477/3e65246a-5d5e-4d6f-96fb-cdae837fecfe">

alisonelizabeth commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into disable_guided_onboarding

xcrzx reviewed (approved):
Rules Area changes LGTM

alisonelizabeth commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into disable_guided_onboarding

CoenWarmer reviewed (approved):
AO changes LGTM

awahab07 reviewed (approved):
Exploratory View changes LGTM!

alisonelizabeth commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into disable_guided_onboarding

alisonelizabeth commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into disable_guided_onboarding

alisonelizabeth commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into disable_guided_onboarding

yuliacech reviewed (approved):
Approving this since we confirmed that the regression in the SIEM guide was caused by an earlier [PR](https://github.com/elastic/kibana/pull/158450).

alisonelizabeth commented:
@stephmilovic do you mind taking another look? I addressed your feedback in https://github.com/elastic/kibana/commit/e6a4568cb5e155028cc2bbed9497c75ad52e8f58.

stephmilovic reviewed (approved):
Sorry for the delay, nasty cold this week 🤧 

LGTM!

alisonelizabeth committed:
Merge branch 'main' of github.com:elastic/kibana into disable_guided_onboarding
